### ISSUE TITLE
Conflict between SA1008 and SA1009 for single-line if

### ISSUE THREAD

**Issue by sharwell** [DotNetAnalyzers/StyleCopAnalyzers#2475]
**Input:**

```csharp
if (true) (true ? 1 : 0).ToString();
```

**Expected result:**

Neither SA1008 nor SA1009 is reported.

**Actual result:**

SA1009 is reported for the closing parenthesis of the `if` condition. Applying the code fix produces the following:

```csharp
if (true)(true ? 1 : 0).ToString();
```

After applying the code fix, SA1008 is reported on the opening parenthesis of the parenthesized ternary expression. Applying the code fix for that restores the original code.
